autoPyTorch/pipeline/components/setup/network_embedding/base_network_embedding.py

```diff
@@ -31,22 +31,37 @@ def transform(self, X: Dict[str, Any]) -> Dict[str, Any]:
     def build_embedding(self, num_input_features: np.ndarray, num_numerical_features: int) -> nn.Module:
         raise NotImplementedError
 
-    def _get_args(self, X: Dict[str, Any]) -> Tuple[None, None]:  # Tuple[int, np.ndarray]:
+    def _get_required_info_from_data(self, X: Dict[str, Any]) -> Tuple[int, np.ndarray]:
+        """
+        Returns the number of numerical columns after preprocessing and
+        an array of size equal to the number of input features
+        containing zeros for numerical data and number of categories
+        for categorical data. This is required to build the embedding.
+
+        Args:
+            X (Dict[str, Any]):
+                Fit dictionary
+
+        Returns:
+            Tuple[int, np.ndarray]:
+                number of numerical columns
+        """
         # Feature preprocessors can alter numerical columns
-        # if len(X['dataset_properties']['numerical_columns']) == 0:
-        #     num_numerical_columns = 0
-        # else:
-        #     X_train = copy.deepcopy(X['backend'].load_datamanager().train_tensors[0][:2])
-        #
-        #     numerical_column_transformer = X['tabular_transformer'].preprocessor. \
-        #         named_transformers_['numerical_pipeline']
-        #     num_numerical_columns = numerical_column_transformer.transform(
-        #         X_train[:, X['dataset_properties']['numerical_columns']]).shape[1]
-        # num_input_features = np.zeros((num_numerical_columns + len(X['dataset_properties']['categorical_columns'])),
-        #                               dtype=int)
-        # categories = X['dataset_properties']['categories']
-        #
-        # for i, category in enumerate(categories):
-        #     num_input_features[num_numerical_columns + i, ] = len(category)
-        # return num_numerical_columns, num_input_features
-        return None, None
+        if len(X['dataset_properties']['numerical_columns']) == 0:
+            num_numerical_columns = 0
+        else:
+            X_train = copy.deepcopy(X['backend'].load_datamanager().train_tensors[0][:2])
+
+            numerical_column_transformer = X['tabular_transformer'].preprocessor. \
+                named_transformers_['numerical_pipeline']
+            num_numerical_columns = numerical_column_transformer.transform(
+                X_train[:, X['dataset_properties']['numerical_columns']]).shape[1]
+
+        num_cols = num_numerical_columns + len(X['dataset_properties']['categorical_columns'])
+        num_input_feats = np.zeros(num_cols, dtype=np.int32)
```

Comment:
In the tabular_feature_validator, we move categoricals to left, but it moves numericals to left.
Doesn't that matter?

Reply:
yeah that is a good point but [here](https://github.com/automl/Auto-PyTorch/blob/e6c37a030223c9a22ec64be723577d0a2e68e4b0/autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/TabularColumnTransformer.py#L61) sklearn moves them to the left. Basically, whichever columns are processed first in a column transformer, are moved to the left.

Reply:
It sounds more efficient in terms of cache efficiency.
Thanks for the info.